rita_common/src/usage_tracker/mod.rs

```diff
@@ -116,8 +117,47 @@ pub struct UsageTracker {
     payments: VecDeque<PaymentHour>,
 }
 
+fn check_usage_hour(input: &mut VecDeque<UsageHour>, last_save_hour: u64) -> bool {
+    let number_of_bytes = 10000;
+    let mut total_used_bytes = 0;
+    while !input.is_empty() {
+        match input.pop_front() {
+            Some(val) => {
+                total_used_bytes += val.up;
+                if val.index < last_save_hour {
+                    return total_used_bytes < number_of_bytes;
+                }
+            }
+            None => break,
+        }
+    }
+    true
+}
+
+fn at_least_two(a: bool, b: bool, c: bool) -> bool {
+    if a {
+        b || c
+    } else {
+        b && c
+    }
+}
+
 impl UsageTracker {
     fn save(&mut self) -> Result<(), RitaCommonError> {
+        let minimum_number_transactions = 75;
+        if self.payments.len() < minimum_number_transactions
+            || at_least_two(
+                check_usage_hour(&mut self.client_bandwidth, self.last_save_hour),
```

Comment:
this should also be called from the saving loop and the number of tx's before a save should be an argument there.

Reply:
The saving loop is really far abstracted from actually touching the transactions. There wasn't a clean way to count the total transactions and prevent bad saves.

Reply:
the transaction counting logic is fine, but this save function should be called in the saving loop.